swim/disseminator.go

```diff
@@ -109,63 +109,88 @@ func (d *disseminator) FullSync() (changes []Change) {
 	return changes
 }
 
-func (d *disseminator) IssueAsSender() []Change {
-	return d.issueChanges(nil)
+// IssueAsSender collects all changes a node needs when sending a ping or
+// ping-req. The second return value is a callback that raises the piggyback
+// counters of the given changes.
+func (d *disseminator) IssueAsSender() (changes []Change, bumpPiggybackCounters func([]Change)) {
+	return d.issueChanges(), d.bumpPiggybackCounters
```

Comment:
Can we return a `func()` here that takes the changes from `d.issueChanges()` and feeds them into `d.bumpPiggybackCounters`. This makes it less error prone on the calling site after the response is received.